dts/bindings/flash_controller/nuvoton,npcx-fiu-nor.yaml

```diff
@@ -28,9 +28,15 @@ properties:
     type: int
     required: true
     description: Mapped memory address of direct read access for spi nor flash.
-  min-erase-size:
+  max-erase-size:
     type: int
     default: 0x10000
+    description: Maximum erase size of spi nor flash.
+    enum:
+      - 0x10000 # 64KB (Block Erase)
+  min-erase-size:
+    type: int
+    default: 0x1000
     description: Minimum erase size of spi nor flash.
     enum:
       - 0x1000  # 4KB (Sector Erase)
```

Comment:
Add ` - 0x10000 # 64KB (Block Erase)`? Or we needn't `BUILD_ASSERT` checks at line 613.

Reply:
If there is a section that needs very less memory but we need to allocate 0x10000 because of erase size, then it causes internal fragmentation right. Having 0x1000 here allows to allocate n * 0x1000 instead of 0x10000 for that section.

I am trying to select a range of erase block, and erasing the memory based on best fit size for the memory that needs to erased with in the configured range. 

With 0x10000 as max and 0x1000 as min we could support 64 KB and 4 KB erases right now. We can also extend it by adding 32 KB erases in future if it's valid and required

Reply:
Build assert are added to make sure following example case won't occur

min-erase-size: 1001
max-erase-size: 2000
Memory size that needs to be erased: 2002
`while` will step into infinite loop
-> first erase - 2000 (2002 - 2000 = 2)
-> 2 is not a valid erase size

Reply:
If you set `min-erase-size` to  0x1001, you will encounter a devicetree error since it is not in the allowed list.

Here is my building result if I set `min-erase-size` to  0x1001.
```
devicetree error: value of property 'min-erase-size' on /soc/quadspi@40021000/w25q256@0 in 
/home/ianchao/zephyrproject/npcx-tests/app/nor_flash/build/zephyr/zephyr.dts.pre (4097) is not in 'enum' list in 
/home/ianchao/zephyrproject/zephyr/dts/bindings/flash_controller/nuvoton,npcx-fiu-nor.yaml ([4096, 65536])
```

Reply:
yes I mean 1001 is an example, but in future if we add more options to enum, this build check will make sure the max is a multiple of min

Reply:
Removed min and max erase size, thank you